modules/space/rendering/renderabletravelspeed.cpp

```diff
@@ -0,0 +1,326 @@
+        "TargetNode",
+        "Target object",
+        "This value sets which scene graph node to target with the light speed indicator"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineColorInfo = {
+        "Color",
+        "Color",
+        "This value determines the RGB color for the line."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineOpacityInfo = {
+        "Opacity",
+        "Opacity",
+        "This value determines the opacity for the line."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineWidthInfo = {
+        "LineWidth",
+        "Line Width",
+        "This value specifies the line width."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo IndicatorLengthInfo = {
+        "IndicatorLength",
+        "Indicator Length",
+        "This value specifies the length of the light indicator set in light seconds."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo FadeLengthInfo = {
+        "FadeLength",
+        "Fade Length",
+        "This value specifies the length of the faded tail of the light indicator "
+        "set in light seconds."
+    };
+
+    struct [[codegen::Dictionary(RenderableLightTravel)]] Parameters {
+        // [[codegen::verbatim(TargetInfo.description)]]
+        std::string target;
+
+        // [[codegen::verbatim(SpeedInfo.description)]]
+        std::optional<double> travelSpeed;
+
+        // [[codegen::verbatim(LineColorInfo.description)]]
+        std::optional<glm::vec3> color;
+
+        // [[codegen::verbatim(LineOpacityInfo.description)]]
+        std::optional<float> opacity;
+
+        // [[codegen::verbatim(LineWidthInfo.description)]]
+        std::optional<float> lineWidth;
+        
+        // [[codegen::verbatim(IndicatorLengthInfo.description)]]
+        std::optional<int> indicatorLength;
+        
+        // [[codegen::verbatim(FadeLengthInfo.description)]]
+        std::optional<int> fadeLength;
+    };
+#include "renderabletravelspeed_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderableTravelSpeed::Documentation() {
+    return codegen::doc<Parameters>("base_renderable_renderabletravelspeed");
+}
+
+RenderableTravelSpeed::RenderableTravelSpeed(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _targetName(TargetInfo)
+    , _travelSpeed(
+        SpeedInfo,
+        distanceconstants::LightSecond,
+        1.0, 
+        distanceconstants::LightSecond
+      )
+    , _indicatorLength(IndicatorLengthInfo, 1, 1, 360)
+    , _fadeLength(FadeLengthInfo, 1, 0, 360)
+    , _lineColor(LineColorInfo, glm::vec3(1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _opacity(LineOpacityInfo, 1.f, 0.f, 1.f)
+    , _lineWidth(LineWidthInfo, 2.f, 1.f, 20.f)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+    setRenderBin(RenderBin::Overlay);
+
+    _lineColor = p.color.value_or(_lineColor);
```

Comment:
This should have a view option for color so that it gets the color picker in the UI. Add the line: 
`_lineColor.setViewOption(properties::Property::ViewOptions::Color);`